__tests__/app.test.js

```diff
@@ -159,3 +159,54 @@ describe("GET: /api/reviews/:review_id/comments", () => {
       });
   });
 });
+
+describe("POST: /api/reviews/:review_id/comments", () => {
+  test("should respond with 201 and posted comment object", () => {
+    return request(app)
+      .post("/api/reviews/2/comments")
+      .send({ username: "bainesface", body: "Not epic at all" })
+      .expect(201)
+      .then((response) => {
+        const commentObj = response.body;
+        expect(commentObj).toBeInstanceOf(Object);
+        expect(commentObj).toHaveProperty("comment");
+        expect(commentObj.comment.length).toBe(1);
+        const expectedObj = {
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          review_id: expect.any(Number),
+          author: "bainesface",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        };
+        expect(commentObj.comment[0]).toMatchObject(expectedObj);
+      });
+  });
+  test("should respond with 404 if review ID does not exist", () => {
+    return request(app)
+      .post("/api/reviews/4234234/comments")
+      .send({ username: "bainesface", body: "Not epic at all" })
+      .expect(404)
+      .then((err) => {
+        expect(err.body.message).toBe("Sorry, review ID not found");
+      });
+  });
+  test("should respond with 400 if review ID is invalid", () => {
+    return request(app)
+      .post("/api/reviews/mushrooms/comments")
+      .send({ username: "bainesface", body: "Not epic at all" })
+      .expect(400)
+      .then((err) => {
+        expect(err.body.message).toBe("Invalid review ID provided");
+      });
+  });
+  test("should respond with 404 if username does not exist", () => {
+    return request(app)
+      .post("/api/reviews/2/comments")
+      .send({ username: "ralphwiggum", body: "I'm learnding" })
+      .expect(404)
+      .then((err) => {
+        expect(err.body.message).toBe(`Sorry, user ID not found`);
+      });
+  });
+});
```

Comment:
Could we just add another couple of tests here please? One for a 201 which ignores unnecessary properties, then one for a 400 when there are missing properties (i.e. no username or body)